From: Kefeng Wang <wangkefeng.wang@huawei.com>
Subject: mm: migrate: remove isolate_movable_page()
Date: Mon, 26 Aug 2024 12:01:32 +0800

There are no more callers of isolate_movable_page(), remove it.

Link: https://lkml.kernel.org/r/20240826040132.1202297-5-wangkefeng.wang@huawei.com
Signed-off-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Cc: Baolin Wang <baolin.wang@linux.alibaba.com>
Cc: David Hildenbrand <david@redhat.com>
Cc: Matthew Wilcox (Oracle) <willy@infradead.org>
Cc: Zi Yan <ziy@nvidia.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/migrate.h |    3 ---
 mm/migrate.c            |    8 --------
 2 files changed, 11 deletions(-)

--- a/include/linux/migrate.h~mm-migrate-remove-isolate_movable_page
+++ a/include/linux/migrate.h
@@ -69,7 +69,6 @@ int migrate_pages(struct list_head *l, n
 		  unsigned long private, enum migrate_mode mode, int reason,
 		  unsigned int *ret_succeeded);
 struct folio *alloc_migration_target(struct folio *src, unsigned long private);
-bool isolate_movable_page(struct page *page, isolate_mode_t mode);
 bool folio_isolate_movable(struct folio *folio, isolate_mode_t mode);
 bool isolate_folio_to_list(struct folio *folio, struct list_head *list);
 
@@ -91,8 +90,6 @@ static inline int migrate_pages(struct l
 static inline struct folio *alloc_migration_target(struct folio *src,
 		unsigned long private)
 	{ return NULL; }
-static inline bool isolate_movable_page(struct page *page, isolate_mode_t mode)
-	{ return false; }
 static inline bool folio_isolate_movable(struct folio *folio,
 					 isolate_mode_t mode)
 	{ return false; }
--- a/mm/migrate.c~mm-migrate-remove-isolate_movable_page
+++ a/mm/migrate.c
@@ -128,14 +128,6 @@ out:
 	return false;
 }
 
-bool isolate_movable_page(struct page *page, isolate_mode_t mode)
-{
-	if (PageTail(page))
-		return false;
-
-	return folio_isolate_movable((struct folio *)page, mode);
-}
-
 static void putback_movable_folio(struct folio *folio)
 {
 	const struct movable_operations *mops = folio_movable_ops(folio);
_
